### ISSUE TITLE
Tab Label centering issues in G5.1.1

### ISSUE THREAD

**Issue by OregSamSas** [MrAlex94/Waterfox#2955]
### What happened?

- [x]  1. I've just upgraded to version **G5.1.1** and modified my Look & feel in preferences, then noticed that a visual glitch (_or I'm not sure how to call it, I feel it's already technical jargon for me, and I'm not good at that (especially not in English)_) when too many tabs were opened so the tabs got exactly as wide that the tab label touched the x mark ("close tab" button). When the label fits on the tab, or is longer (so its end is hidden), nothing (visually irritating) happens. <html><details><summary>GIF</summary><img src="https://user-images.githubusercontent.com/101597941/210565043-8d41b930-b225-4dde-b650-6c06d5c10fb1.gif" alt="Tab glitch"></img></details><details><summary>Video</summary>https://user-images.githubusercontent.com/101597941/210564937-68f614a1-d72f-4701-b26d-cf9b37f5c5d0.mp4</details></html>

- [ ] 2. Furthermore, some of the tab labels wobbled (_idk the right word for it_) on **hovering them**. I'll attach an animated GIF that surely explains it better than me. <html><details><summary>GIF</summary><img src="https://user-images.githubusercontent.com/101597941/210570709-e53c4684-1618-41fd-8eda-464854957246.gif" alt="Tab label wobbling"></img></details></html>

Later, I realized that these must have something to do with the **tab centring setting** under "**Look & Feel**", because when I disabled it, everything worked fine.

I did not test it on FireFox since I don't have it installed.

### Reproducible?

- [ ] I have tested this on Firefox.

### Version

Other

### What platform are you seeing the problem on?

Windows

**Comment by OregSamSas** [comment]
I'd appreciate if someone could check it on FireFox

**Comment by DavidSchillinger** [comment]
Just wanted to confirm that this still happens on G5.1.2 (Windows 11).
I'd like to add that these issues don't seem to happen consistently for me.
Restarting Waterfox seems to fix the issue temporarily for me.
Sometimes I need to disable/re-enable the setting (Center Tab Content) a couple of times and suddenly the glitches are back.

**Comment by OregSamSas** [comment]
> I'd like to add that these issues don't seem to happen consistently for me.

Yes, on G5.1.2 it doesn't happen to me in every case anymore either (only randomly), so it's not disturbing. Just curious though, if you keep toggling it, somehow the glitches show up again, as you said:

> Sometimes I need to disable/re-enable the setting (Center Tab Content) a couple of times and suddenly the glitches are back.

> Restarting Waterfox seems to fix the issue temporarily for me.

Restarting solves the problem for me as well. And following that, I haven't met reappearing unless after interfering with the option in the settings.

> (Windows 11)

By the way, I have Windows 10, if that can be a problem.

**Comment by OregSamSas** [comment]
Or opening a specific number of tabs also leads to glitch No. 2, but not on every tabs. I haven't met with the first glitch since I posted this issue. So that part might have been solved utterly.
